src/jrd/replication/Publisher.cpp

```diff
@@ -303,21 +347,26 @@ void REPL_attach(thread_db* tdbb, bool cleanupTransactions)
 	const auto dbb = tdbb->getDatabase();
 	const auto attachment = tdbb->getAttachment();
 
-	const auto replMgr = dbb->replManager();
-	if (!replMgr)
+	const auto replConfig = dbb->replConfig();
+	if (!replConfig)
 		return;
 
 	fb_assert(!attachment->att_repl_matcher);
 	auto& pool = *attachment->att_pool;
-	attachment->att_repl_matcher = replMgr->createTableMatcher(pool);
+	attachment->att_repl_matcher = FB_NEW_POOL(pool)
+				TableMatcher(pool, replConfig->includeFilter, replConfig->excludeFilter);
 
```

Comment:
Do you intend to use include/exclude filters for custom plugins too? If/When plugin name is moved from the config file to RDB$PUBLICATIONS, then I'd expect that replication plugins don't need replication.conf at all (plugin may provide its own filters, if necessary).

Reply:
If there's no such an intention, then the matcher does not make any sense before the manager is created. If you want to defer manager creation, then the matcher should be created inside getReplicator() together with the replicator itself. In this case, I see no sense to create the matcher from config rather than from the manager. And all the changes around replManager() and replConfig() look unnecessary.

Reply:
I have no idea why you didn't remove the matcher after making including tables into publication in SQL so preserved your logic as much as possible.

If you wish, I'll drop the matchers completely.

Reply:
Most of changes around replManager() caused by fact that it is useful only for internal replication. Additional thread that is created inside is a waste of resources for any external plugin.

Reply:
Regexps are handy if you need to exclude a subset of tables, so I decided to leave the matcher to have an override option (second level of customization). So far I don't want to drop it.

Reply:
I agree that regexps are handy. They should be supported in ALTER PUBLICATION statement.